spec/views/post_show_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe 'posts/show.html.erb', type: :view do
+  describe 'a user show page' do
+    before(:each) do
+      @user = User.create(name: 'Motunrayo',
+                          photo: 'https://picsum.photos/200',
+                          bio: 'Abuja Real Lady',
+                          posts_counter: 0)
+      @first_post = Post.create(author_id: @user.id, title: 'Hello 1', text: 'This is my first post',
+                                comments_counter: 0, likes_counter: 0)
+      @last_post = Post.create(author_id: @user.id, title: 'Hello 4', text: 'This is my fourth post',
+                               comments_counter: 0, likes_counter: 0)
+      Comment.create(post_id: @last_post.id, author_id: @user.id, text: 'This is the 1st comment on the 4th post')
+      Comment.create(post_id: @last_post.id, author_id: @user.id, text: 'This is the 2st comment on the 4th post')
+      Comment.create(post_id: @last_post.id, author_id: @user.id, text: 'This is the 3st comment on the 4th post')
+      visit user_posts_path(@user)
+    end
+    it 'shows user name' do
+      expect(page).to have_content(@user.name)
+    end
+    it 'shows the user photo' do
+      expect(page.body).to include('https://picsum.photos/200')
+    end
+    it 'shows number of comments a post has' do
+      expect(page.body).to include('Comments: 3')
+    end
+    it 'shows number of likes a post has' do
+      expect(page.body).to include('Likes: 0')
+    end
+    it 'shows the title of the post' do
+      expect(page).to have_content(@last_post.title)
+    end
+    it 'shows part of a post body (This is my first post)' do
+      expect(page).to have_content('This is my first post')
+    end
+    it 'Not to show the Pagination button' do
+      expect(page.body).not_to include('Pagination')
+    end
+    it 'Redirect to post show page when a post is clicked' do
+      click_link(@last_post.title)
+      expect(current_path).to eq(user_post_path(@user, @last_post))
+    end
+  end
```

Comment:
- Just as i have pointed out above, some of the required tests are missing here too. I'll suggest you carefully go through the requirements and include all. :+1: :wave: 
- - Post show requirements:
![Selection_073](https://github.com/Ademola-coding/Blog_rails/assets/107632780/87c5e911-a54e-41e9-ae29-80c49d14bed6)

- [ ] this test `    it 'shows part of a post body (This is my first post)' do
      expect(page).to have_content('This is my first post')
    end` should be checking that the post body is visible and not just part of the body. I'll suggest you update this. :+1: :100: 

- [ ] Also this is already the post show page, this test `'Redirect to post show page when a post is clicked'` is redundant here. :+1: :100:

Reply:
![Screenshot (134)](https://github.com/Ademola-coding/Blog_rails/assets/96092850/81c29be1-1eb1-4cad-ad17-9463ed2bd729)

made proper corrections